Server/JokesOnYou.Web.Api/Repositories/UserRepository.cs

```diff
@@ -36,32 +47,69 @@ public async Task<User> CreateUserAsync(UserRegisterDTO userRegisterDTO)
                 }
 
                 throw new UserRegisterException(message.ToString());
-            }
+            }   
         }
 
-        public async Task DeleteUserAsync(User user)
+        public async Task DeleteUserAsync(string id)
         {
+            var user = await GetUserAsync(id);
+            if (user == null)
+                throw new Exception($"Cant find user of id:{id}");
             await _userManager.DeleteAsync(user);
         }
 
-        public Task<User> GetUserAsync(string id)
+        public async Task<UserReplyDTO> GetUserReplyAsync(string id)
         {
             //Possibilities 
             //var user = await _userManager.FindByIdAsync(id);
             //var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Id == id);
-            return _userManager.FindByIdAsync(id);
+            var user = await _userManager.FindByIdAsync(id);
+            
+            return _mapper.Map<UserReplyDTO>(user);
+        }
+        public async Task<User> GetUserAsync(string id)
+        {
+            //Possibilities 
+            //var user = await _userManager.FindByIdAsync(id);
+            //var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Id == id);
+            return await _userManager.FindByIdAsync(id);
         }
-
         public Task<User> GetUserByEmail(string email)
         {
             return _userManager.FindByEmailAsync(email);
         }
 
-        public async Task<IEnumerable<User>> GetUsersAsync()
+        public async Task<IEnumerable<UserReplyDTO>> GetUsersAsync()
         {
-            //Does not need to be awaited here.
+            //Such a waste of bits
             var users = await _userManager.Users.ToListAsync();
-            return users;
+            
+            return _mapper.Map<List<UserReplyDTO>>(users);
+        }
+
+        public async Task UpdateUser(User user)
```

Comment:
I'm not sure how this works with mapping and saving ? do you have any idea @N1c0z or @chrisK00 ? care to explain to me ?

Reply:
Just looked it up.
This is good.
https://stackoverflow.com/questions/39545037/updating-user-by-usermanager-update-in-asp-net-identity-2